### PR TITLE
2198 Add CenterScaleCrop transform

### LINKED ISSUE #2198: Add `ScaleCenterCrop` transform
**Is your feature request related to a problem? Please describe.**
Need to implement a deterministic transform that center crop the image with a scale value.

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #2198 .

### Description
This PR added the `CenterScaleCrop` transform according to @rijobro 's feature request.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/croppad/dictionary.py [hunk 1]
```diff
@@ -400,6 +403,32 @@ def inverse(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.ndar
...
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: monai.transforms.MapTransform
+        roi_scale: specifies the expected scale of image size to crop. e.g. [0.3, 0.4, 0.5] or a number for all dims.
+            If its components have non-positive values, will use `1.0` instead, which means the input image size.
+        allow_missing_keys: don't raise exception if key is missing.
+    """
+
+    def __init__(
+        self, keys: KeysCollection, roi_scale: Union[Sequence[int], int], allow_missing_keys: bool = False
```
Comment (rijobro) [inline monai/transforms/croppad/dictionary.py:419]:
```suggestion
        self, keys: KeysCollection, roi_scale: Union[Sequence[float], float], allow_missing_keys: bool = False
```

**Review by rijobro — APPROVED** [review]
thanks!

File: monai/transforms/croppad/dictionary.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/croppad/dictionary.py:419]: good catch!
Thanks.

**Comment by jpcenteno80** [comment]
I thought this transform used to have the `inverse` method implemented.  I am testing code from a couple of months ago with this version `MONAI version: 0.5.3+116.gb151b8e` and I run into the following error:
```
  File "/home/jpcenteno/miniconda3/envs/test_pkgs/lib/python3.9/site-packages/monai/transforms/inverse.py", line 121, in inverse
    raise NotImplementedError(f"Subclass {self.__class__.__name__} must implement this method.")
NotImplementedError: Subclass CenterScaleCropd must implement this method.
```

**Comment by Nic-Ma** [comment]
Thanks for your report. Let me take a look.

**Comment by jpcenteno80** [comment]
Thanks. I added some sample code that fixed the issue under this discussion: #2380

**Pull request merged**